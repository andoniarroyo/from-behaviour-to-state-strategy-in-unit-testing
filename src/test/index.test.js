import transform from '../main';

describe('index.js', () => {
  it('transforms the incoming raw object', () => {
    const rawObject = {
      name: 'NAME',
      content: [
        {
          type: 'type1',
          value: 'PROP1',
        },
        {
          type: 'type2',
          value: 'prop2',
        },
      ],
    };

    const result = transform(rawObject);

    // checking by state
    expect(result).toMatchSnapshot();
  });
});