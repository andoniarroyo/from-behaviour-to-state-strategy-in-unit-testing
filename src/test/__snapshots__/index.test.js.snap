// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index.js transforms the incoming raw object 1`] = `
Object {
  "content": Array [
    Object {
      "type": "type1",
      "value": "prop1",
    },
    Object {
      "type": "type2",
      "value": "PROP2",
    },
  ],
  "name": "name",
}
`;
